fossunited/api/schedule.py

```
@@ -11,6 +21,20 @@ def format_date(date_obj):
+def to_time(val):
+    # Accept datetime.time, timedelta, or "HH:MM:SS"/"HH:MM" strings
+    if isinstance(val, datetime.time):
+        return val
+    if isinstance(val, timedelta):
+        return (datetime.min + val).time()
+    if isinstance(val, str):
+        try:
+            return datetime.strptime(val.strip(), "%H:%M:%S").time()
+        except ValueError:
+            return datetime.strptime(val.strip(), "%H:%M").time()
+    raise TypeError(f"Unsupported time type: {type(val)}")
```

Ruff (0.12.2)

Avoid specifying long messages outside the exception class

(TRY003)